types/kagekiri.d.ts

```diff
@@ -4,13 +4,111 @@
+ *
+ * @param names - class name or whitespace-separated class names
+ * @param context - context to query in, or `document` by default
+ */
 export function getElementsByClassName(names: string, context?: Node): Element[];
+
+
+/**
+ * Query for all elements matching a given tag name. Analogous to
+ * [`Document.getElementsByTagName`](https://developer.mozilla.org/en-US/docs/Web/API/Document/getElementsByTagName).
+ * The `"*"` query is supported.
+ *
+ * Unlike the standard API, this returns a static array of Elements rather than a live HTMLCollection.
+ *
+ * The default `context` is `document`. Choose another node to query within that context.
+ *
+ * @param tagName - name of the element tag
+ * @param context - context to query in, or `document` by default
+ */
 export function getElementsByTagName(tagName: string, context?: Node): Element[];
+
+/**
+ * Query for all elements matching a given tag name and namespace. Analogous to
+ * [`Document.getElementsByTagNameNS`](https://developer.mozilla.org/en-US/docs/Web/API/Document/getElementsByTagNameNS).
+ * The `"*"` query is supported.
+ *
+ * Unlike the standard API, this returns a static array of Elements rather than a live NodeList.
+ *
+ * The default `context` is `document`. Choose another node to query within that context.
+ *
+ * @param namespaceURI - namespace URI, or `"*"` for all
+ * @param localName - local name, or `"*"` for all
+ * @param context - context to query in, or `document` by default
+ */
 export function getElementsByTagNameNS(namespaceURI: string, localName: string, context?: Node): Element[];
+
+/**
+ * Query for an element matching the given ID, or null if not found. Analogous to
+ * [`Document.getElementById`](https://developer.mozilla.org/en-US/docs/Web/API/Document/getElementById)
+ *
+ * The default `context` is `document`. Choose another DocumentOrShadowRoot to query within that context.
+ *
+ * @param id - element ID
+ * @param context - context to query in, or `document` by default
+ */
 export function getElementById(id: string, context?: DocumentOrShadowRoot): Element | null;
+
+/**
+ * Query for all elements matching a given name. Analogous to
+ * [`Document.getElementsByName`](https://developer.mozilla.org/en-US/docs/Web/API/Document/getElementsByName)
+ *
+ * The default `context` is `document`. Choose another DocumentOrShadowRoot to query within that context.
+ *
+ * Unlike the standard API, this returns a static array of Elements rather than a live NodeList.
+ *
+ * @param name - element name attribute
+ * @param context - context to query in, or `document` by default
+ */
 export function getElementsByName(name: string, context?: DocumentOrShadowRoot): Element[];
+
+/**
+ * Return true if the given Node matches the given CSS selector, or false otherwise. Analogous to
+ * [`Element.closest`](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest)
+ *
+ * @param selector - CSS selector
+ * @param context - element to match against
```

Comment:
Should this be `@param element`? I went back and forth on that.

Reply:
I would say yes, since it's not really the "context" per se. (Also "context" is not the perfect word; not sure why I thought of it. :) )

In any case, let's change it to `element` for both `closest` and `matches`. That seems uncontroversial enough.